### PR TITLE
Basic RocksDB KVStore implementation

### PR DESCRIPTION
**Pull request by alari**

### CODE REVIEW

File: effects/kvstore/src/main/scala/fluence/effects/kvstore/KVStore.scala [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+package fluence.effects.kvstore
+
+import cats.Monad
+import cats.data.EitherT
+import fluence.codec.PureCodec
+
+import scala.language.higherKinds
+
+abstract class KVStore[F[_]: Monad, K, V] {
+  self ⇒
```
Comment (folex) [inline effects/kvstore/src/main/scala/fluence/effects/kvstore/KVStore.scala]: self-typing without a type?

File: effects/kvstore/src/main/scala/fluence/effects/kvstore/RocksDBStore.scala [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        }
+      )
+
+      // Read options -- could be used for optimizations later, e.g. snapshots
+      readOptions ← Resource.make(IO(new ReadOptions()))(
+        rOpts ⇒
+          IO(rOpts.close()).handleError {
+            case NonFatal(err) ⇒
+              err.printStackTrace()
+              logger.error(s"Cannot close RocksDB object during cleanup: $err", err)
```
Comment (folex) [inline effects/kvstore/src/main/scala/fluence/effects/kvstore/RocksDBStore.scala]: stack trace should be after log

File: effects/kvstore/src/main/scala/fluence/effects/kvstore/RocksDBStore.scala [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        }
+      )
+
+    } yield
+      new KVStore[F, Array[Byte], Array[Byte]] {
```
Comment (folex) [inline effects/kvstore/src/main/scala/fluence/effects/kvstore/RocksDBStore.scala]: Why create anonymous class? Better move it to a named class

File: effects/kvstore/src/main/scala/fluence/effects/kvstore/KVStore.scala (hunk 1 above)
Comment (alari) [reply effects/kvstore/src/main/scala/fluence/effects/kvstore/KVStore.scala]: It's not typing, it's a self-reference, used to build anonymous inner classes with delegating the logic to `self`.

**Pull request merged**